pkg/queue/instance.go

```diff
@@ -117,6 +117,48 @@ func (q *queueImpl) Closed() <-chan struct{} {
 	return q.closed
 }
 
+// Get blocks until it can return a task to be processed. If shutdown = true,
+// the processing go routine should stop.
+func (q *queueImpl) get() (task *BackoffTask, shutdown bool) {
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
+	// wait for closing to be set, or a task to be pushed
+	for !q.closing && len(q.tasks) == 0 {
+		q.cond.Wait()
+	}
+
+	if len(q.tasks) == 0 {
```

Comment:
This is not equal to previous behavior

Reply:
I can change to previous condition but it is same behaviour. Wait only comes out if len(q.tasks) is zero

Reply:
Previously, if queue is stopped, no tasks will be run if there is any left unprocessed.

Reply:
ok. any way I changed to previous condition.